e2e/test/scenarios/question/reproductions/22230-query-builder-aggregation-filtering.cy.spec.js

```diff
@@ -0,0 +1,43 @@
+import {
+  restore,
+  popover,
+  summarize,
+  openPeopleTable,
+  visualize,
+} from "e2e/support/helpers";
+
+describe("issue 22230", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should allow filtering on aggregated fields", () => {
+    openPeopleTable({ mode: "notebook" });
+
+    summarize({ mode: "notebook" });
+    popover().findByRole("option", { name: "Maximum of ..." }).click();
+    // I don't know why this can't be chained, I tried it and it didn't work.
+    popover().findByRole("option", { name: "Name" }).click();
+
+    // findByRole doesn't work here, and I don't know why.
+    cy.findByText("Pick a column to group by").click();
+    popover().findByRole("option", { name: "Source" }).click();
+
+    // can't use `filter()` because it found multiple filter buttons and Cypress only allows clicking on one element at a time.
+    cy.button("Filter").click();
+    popover().findByRole("option", { name: "Max of Name" }).click();
+    // can't add aria-label to the input, since it's past the translation period.
+    cy.findByTestId("select-button").click();
+    // can't use `popover()` because there are 2 popovers, and I tested that it also didn't work.
+    cy.findByRole("option", { name: "Starts with" }).click();
```

Comment:
I've tried `popover().findByRole("option", { name: "Starts with" }).click()`, but it doesn't work. I suspect it has something to do with `popover` finding 2 popovers.